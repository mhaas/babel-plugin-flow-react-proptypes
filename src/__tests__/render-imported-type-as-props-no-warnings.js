const content = `
import React, { Component } from 'react';
import renderer from 'react-test-renderer';

import type { SomeNestedType } from './test-types.js.transpiled';

type Props = SomeNestedType;

class App extends Component {
  props: Props

  render() {
    const { nestedProperty } = this.props;

    return (
      <div className="App">
        <p className="App-intro">
          { JSON.stringify(nestedProperty, null, 4)}
        </p>
      </div>
    );
  }
}

export default App;
`;

const utils = require('./lib/render-component');

/*
 * Test case for 101: Exported object types should just work when used directly as proptypes.
 *
 */

it('imported-object-type-can-be-used-as-props: no warnings', () => {
  const path = require('path');
  const renderCall = 'renderer.create(<App stringValue="Test"/>);';
  const fullSource = content + renderCall;
  const errorsSeen = utils.getConsoleErrorsForComponent(fullSource, [
    [path.join(__dirname, 'lib', 'test-types.js.pre-transpile'),
      path.join(__dirname, 'lib', 'test-types.js.transpiled')]
  ]);
  expect(errorsSeen).toMatchSnapshot();
});
