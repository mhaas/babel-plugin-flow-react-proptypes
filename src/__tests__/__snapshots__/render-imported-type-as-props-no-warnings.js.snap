exports[`test imported-object-type-can-be-used-as-props: no warnings 1`] = `
Array [
  Array [
    "Warning: Failed prop type: The prop \`a\` is marked as required in \`App\`, but its value is \`undefined\`.
    in App",
  ],
  Array [
    "Warning: Failed prop type: The prop \`b\` is marked as required in \`App\`, but its value is \`undefined\`.
    in App",
  ],
  Array [
    "Warning: Failed prop type: The prop \`nestedProperty\` is marked as required in \`App\`, but its value is \`undefined\`.
    in App",
  ],
]
`;
